lib/elastomer/client/percolator.rb

```diff
@@ -0,0 +1,77 @@
+module Elastomer
+  class Client
+
+    class Percolator
+
+      # Create a new Percolator for managing a query.
+      #
+      # client     - Elastomer::Client used for HTTP requests to the server
+      # index_name - The index name
+      # id         - The _id for the query
+      def initialize(client, index_name, id)
+        @client = client
+        @index_name = client.assert_param_presence(index_name, 'index name')
+        @id = id
```

Comment:
Is `id` also a required parameter? Should probably `assert_param_presence` on that too.

Bit of historical background, `assert_param_presence` exists because a nil or empty string parameter can cause a command to apply to more things than you expected. See https://github.com/github/elastomer-client/issues/32.

I haven't checked the api, but it's possible that the request `DELETE /index_name/.percolator/` deletes all percolators. In that case we should definitely require a non-blank `id`.